assert.sh

```diff
@@ -139,6 +139,7 @@ _assert_fail() {
     fi
     tests_errors[$tests_failed]="$report"
     (( tests_failed++ )) || :
+    return 1
```

Comment:
This allows us to check if an assertion has failed or not. For example we can now do:
`assert_raises 'assert "echo foo" "bar"' 1`
That's not exciting, but useful for testing the new `assert_success` assertion:
`assert_raises 'assert_success "false"' 1`